### PR TITLE
Attempt to fix #11756

### LINKED ISSUE #11756: Implement c output format for all t commands
So we can tc > foo.h and then to foo.h

See my previous commits in tu, ts, te.. to handle json

* [x] ttc
* [x] tec
* [x] tsc
* [x] tuc

### PR DESCRIPTION
**Pull request by dav1901**
Implement C output formats to structs and unions.

### CODE REVIEW

File: libr/core/cmd_type.c [hunk 1]
```diff
@@ -748,6 +782,39 @@ static int cmd_type(void *data, const char *input) {
...
+							char *val = sdb_array_get (TDB, var2, 0, NULL);
+							if(val){
+								r_cons_printf("%s%s %s;",space,val,p);
+								space = " ";
+							}
+							free(val);
+						}
+						free(var2);
+					}
+					free(var);
```
Comment (radare) [inline libr/core/cmd_type.c]: add a space before every (

File: libr/core/cmd_type.c [hunk 2]
```diff
@@ -655,6 +657,38 @@ static int cmd_type(void *data, const char *input) {
...
+					free(name);
+					name = strdup (sdbkv_key (kv));
+					r_cons_printf ("%s %s {", sdbkv_value (kv), name);
+					char *p,*var = r_str_newf ("%s.%s",sdbkv_value(kv), name);
+					for(int n = 0; (p = sdb_array_get (TDB, var, n, NULL)); n++){
+						char *var2 = r_str_newf ("%s.%s",var,p);
+						if(var2){
+							char *val = sdb_array_get (TDB, var2, 0, NULL);
+							if(val){
+								r_cons_printf("%s%s %s;",space,val,p);
```
Comment (radare) [inline libr/core/cmd_type.c]: spaces after commas

File: libr/core/cmd_type.c [hunk 3]
```diff
@@ -748,6 +782,39 @@ static int cmd_type(void *data, const char *input) {
 		case 0:
 			print_keys (TDB, core, stdifstruct, printkey_cb, false);
 			break;
+
+		case 'c':{
+			char *name = NULL;
+			SdbKv *kv;
+			SdbListIter *iter;
+			SdbList *l = sdb_foreach_list_filter (TDB, stdifstruct, true);
+			const char *space = "";
```
Comment (radare) [inline libr/core/cmd_type.c]: reduce nestness, use continue with early conditionals

**Comment by radare** [comment]
dupe of https://github.com/radare/radare2/pull/12408

**Comment by radare** [comment]
dupe of https://github.com/radare/radare2/pull/12408

**Comment by dav1901** [comment]
Should I close this in favor of #12408 ?

**Comment by radare** [comment]
You are implement tuc and the other only tsc. I think both prs have good things but will be good to not overlap work

File: libr/core/cmd_type.c [hunk 4]
```diff
@@ -655,6 +657,38 @@ static int cmd_type(void *data, const char *input) {
...
 			}
 			break;
+		case 'c':{
+			char *name = NULL;
+			SdbKv *kv;
+			SdbListIter *iter;
+			SdbList *l = sdb_foreach_list_filter (TDB, stdifunion, true);
+			const char *space = "";
+			ls_foreach (l, iter, kv) {
+				if(!name || strcmp (sdbkv_value (kv), name)){
```
Comment (XVilka) [inline libr/core/cmd_type.c]: Can you put the logic in a separate function to overcome such a deep nesting?

**Pull request merged**